tests/framework/Optimizers/GradientDescent/gold/ConstrainFunction/opt_export_0.csv

```diff
@@ -10,190 +10,190 @@ x,y,ans,stepSize,iteration,accepted,conv_gradient,rejectReason
 0.750001023127,0.579721140494,1.18452256815,0.211140874198,8.0,accepted,0.0,
 0.883326689385,0.118735960603,1.01477679852,0.239939090897,9.0,accepted,0.0,
 0.328839088104,-0.0056564168228,0.192803675172,0.284134635136,10.0,accepted,0.0,
--0.221993457008,-0.131320515084,0.0158633782179,0.282492459345,11.0,rejected,0.0,2
+-0.221993457008,-0.131320515084,0.0158633782179,0.282492459345,11.0,rejected,0.0,implicitConstraintsViolation
 0.328839088104,-0.0056564168228,0.192803675172,0.282492459345,12.0,rerun,0.0,
 -0.173326792157,-0.113534397614,0.00555999836688,0.256811326677,13.0,accepted,0.0,
-0.435740084796,-0.0191403153051,0.293555727067,0.308169070227,14.0,rejected,0.0,1
+0.435740084796,-0.0191403153051,0.293555727067,0.308169070227,14.0,rejected,0.0,noImprovement
 -0.173326792157,-0.113534397614,0.00555999836688,0.308169070227,15.0,rerun,0.0,
 -0.1690493633,-0.112604211161,0.00492668071114,0.280153700206,16.0,accepted,0.0,
```

Comment:
this is not standard... the number of columns varies from row to row.

Reply:
I changed the none into a string, hope this could be ok.